SIPS/sip-29.md

```diff
@@ -243,13 +269,138 @@ type FungibleAssetMetadata = {
   // List of asset units.
   units: FungibleAssetUnit[];
 };
+```
+### Asset Market Data
+
+```typescript
+type FungibleAssetMarketData = {
+  // Represents a fungible asset market data.
+  fungible: true;
+
+  // The market cap of the asset represented as a decimal number in a string.
+  marketCap?: string;
+
+  // The total volume of the asset represented as a decimal number in a string.
+  totalVolume?: string;
+
+  // The circulating supply of the asset represented as a decimal number in a string.
+  circulatingSupply?: string;
+
+  // The all time high of the asset represented as a decimal number in a string.
+  allTimeHigh?: string;
+
+  // The all time low of the asset represented as a decimal number in a string.
+  allTimeLow?: string;
+
+  pricePercentChange?: {
+    // The `all` value is a special interval that represents all available data.  
+    all?: number;  
+
+    // The interval key MUST follow the ISO 8601 duration format.  
+    [interval: string]: number;
+  };
+};
+```
+
+## Appendix II: Non-fungible Asset Metadata & Market Data
+
+The following asset metadata fields for a non-fungible asset are defined.
+
+### Asset Metadata
 
-// Represents the metadata of an asset.
-type AssetMetadata = FungibleAssetMetadata
+`Caip10Address` - A string that represents an address using the [CAIP-10][caip-10] standard.
+
+```typescript
+type NonFungibleAssetCollection = {
+  // Human-friendly name of the asset collection.
+  name: string;
+
+  // Ticker symbol of the asset collection.
+  symbol: string;
+
+  // The number of tokens in the collection.
+  tokenCount: string;
+
+  // The creator address of the asset.
+  creator?: Caip10Address;
+
+  // Base64 data URI or URL representation of the asset icon.
+  imageUrl?: string;
+};
+
+type NonFungibleAssetMetadata = {
+  // Human-friendly name of the asset.
+  name?: string;
+
+  // Ticker symbol of the asset.
+  symbol?: string;
+
+  // Base64 data URI or URL representation of the asset image.
+  imageUrl?: string;
+
+  // The description of the asset.
+  description?: string;
+
+  // Represents a non-fungible asset
+  fungible: false;
+
+  // The time at which the asset was acquired.
+  // The time is represented as a UNIX timestamp. 
+  acquiredAt?: number;
+
+  // Attributes of the non-fungible asset.
+  attributes?: Record<string, string | number>;
+
+  // The collection of the asset.
+  collection?: NonFungibleAssetCollection;
+};
+```
+
+### Asset Market Data
+```typescript
+type NonFungibleAssetMarketData = {
+  // Represents a non-fungible asset market data.
+  fungible: false;
+
+  // The last sale of one asset in the collection.
+  lastSale?: {
+    // The asset that was sold.
+    asset: Caip19AssetTypeOrId;
```

Comment:
I see the asset returned by the nft-api looks like this
`{
            "type": "native",
            "name": "Wrapped SOL",
            "symbol": "SOL",
            "decimals": 9,
            "token_id": "So11111111111111111111111111111111111111112"
          }`
What would the snap return in this case?

Reply:
The snap would use its assets data to get the SOL CAIP-19 ID and return that. The data that the API returns is not CAIP compliant so we need to do an extra step to convert it :) Overall the idea is not just do dump bind our API to the Snap API as it's clearly not the intention of the Snaps APIs but the snap itself can deal with transforming the data to the expected shape